### PR TITLE
Make gdbserver great again

### PR DESCRIPTION
**Pull request by bet4it**
- [x] Have you checked that there aren't other open [pull requests](https://github.com/qilingframework/qiling/pulls) for the same update/change?
- [x] Is this pull request for [dev](https://github.com/qilingframework/qiling/tree/dev) branch? (Do not pull request for **master** branch).

-----

`reg_map` was used when gdbserver send all the registers, registers like `w0` shouldn't be included in `reg_map`

### CODE REVIEW

File: qiling/arch/arm64_const.py [hunk 1]
```diff
@@ -37,6 +37,14 @@
...
             "x29": UC_ARM64_REG_X29,
             "x30": UC_ARM64_REG_X30, 
+            "sp": UC_ARM64_REG_SP,
+            "pc": UC_ARM64_REG_PC,
+            "lr": UC_ARM64_REG_LR,
+            "cpacr_el1": UC_ARM64_REG_CPACR_EL1,
+            "tpidr_el0": UC_ARM64_REG_TPIDR_EL0,
+}
+
+reg_map_part = {
```
Comment (xwings) [inline qiling/arch/arm64_const.py:47]: this changes is gdb specific. gdb should handle the map and split.

Comment (bet4it) [reply qiling/arch/arm64_const.py:47]: This regression was introduced in ef0ac2a39ac408d2b61ba235894591ad8fdd7e25

Comment (xwings) [reply qiling/arch/arm64_const.py:47]: What i am saying is, the split should be done by gdb. not the main constant

Comment (bet4it) [reply qiling/arch/arm64_const.py:47]: `reg_map` is used to generate `self.tables`:
https://github.com/qilingframework/qiling/blob/d59436e673a0f7feaf3865cf0fb9245d4ec0d3a5/qiling/debugger/gdb/gdb.py#L91-L98
The order of registers in `self.tables` should be same with their order in gdb, which is used in:
https://github.com/qilingframework/qiling/blob/d59436e673a0f7feaf3865cf0fb9245d4ec0d3a5/qiling/debugger/gdb/gdb.py#L234
https://github.com/qilingframework/qiling/blob/d59436e673a0f7feaf3865cf0fb9245d4ec0d3a5/qiling/debugger/gdb/gdb.py#L292
https://github.com/qilingframework/qiling/blob/d59436e673a0f7feaf3865cf0fb9245d4ec0d3a5/qiling/debugger/gdb/gdb.py#L379
https://github.com/qilingframework/qiling/blob/d59436e673a0f7feaf3865cf0fb9245d4ec0d3a5/qiling/debugger/gdb/gdb.py#L435
We could handle the split in these places, but the changes will be will ugly and unnecessary.

Comment (xwings) [reply qiling/arch/arm64_const.py:47]: make sense,

and again, what does reg_map_part means ? maybe a better na,e ?

Comment (bet4it) [reply qiling/arch/arm64_const.py:47]: `part` represents `partial registers` or just `part of a register`
I'm not sure how to call registers like `eax` or `w0` in a better way
I thought of name like `reg_map_lb`, where `lb` represents `lower bits`, but I don't know if it's proper to use this on registers like `ah`

Comment (xwings) [reply qiling/arch/arm64_const.py:47]: extend sounds better
reg_map_extend

Comment (bet4it) [reply qiling/arch/arm64_const.py:47]: Do you mean `w0` is an extend of `x0`? I don't think it's suitable.

Comment (xwings) [reply qiling/arch/arm64_const.py:47]: its a extension of the current main set

Comment (bet4it) [reply qiling/arch/arm64_const.py:47]: That's up to you, you can change it by yourself😄

Comment (xwings) [reply qiling/arch/arm64_const.py:47]: your PR :) once you update i will merge.

**Comment by xwings** [comment]
I will do it :)

**Pull request merged**